AuthorRank/graph.py

```diff
@@ -0,0 +1,90 @@
+# imports
+from collections import Counter
+import itertools
+import networkx as nx
+from typing import List
+
+
+def create(documents: List[dict], authorship_key: str = "authors", keys: set = None) -> 'nx.classes.digraph.DiGraph':
+
+    """
+    Creates a directed graph object from the list of input documents which are represented as dictionaries.
+    :param documents: a list of dictionaries which represent documents.
+    :param authorship_key: the key in the document which contains a list of dictionaries representing authors.
+    :param keys: a set that contains the keys to be used to create a UID for authors.
+    :return: a networkx DiGraph object.
+    """
+
+    # if keys are not provided, set a default
+    # see https://florimond.dev/blog/articles/2018/08/python-mutable-defaults-are-the-source-of-all-evil/
+    if keys is None:
+        keys = {"first_name", "last_name"}
+
+    # get the authorship from each of the documents
+    documents = [i[authorship_key] for i in documents]
+
+    # remove keys and values that are not used as part of an author UID
+    for doc in documents:
+        for author in doc:
```

Comment:
I think the name author on line 28 should be refactored. It's misleading as not all of the keys will be authors.